data/scenarios/Challenges/Mazes/easy_spiral_maze.yaml

```diff
@@ -0,0 +1,104 @@
+name: Spiral shaped maze
+description: A maze shaped like a spiral, with a twist!
+objectives:
+- goal:
+  - You find yourself in the middle of a large maze.
+  - It's straightforward to get out, but the path is long and dull.
+  - You need to send a robot to the goal square, labelled with an exclamation mark;
+    you win by `grab`bing the `goal`.
+  - Beware! The winding corridors are wider then they look!
+  condition: |
+    j <- robotNamed "judge";
+    as j {has "goal"}
+solution: |
+  def until = \c. b <- c; if b {} {until c} end;
+  build {
+    log "I am on my way!";
+    turn west;
+    until (
+      b <- blocked;
+      b <- if b { turn right; blocked } { return false };
+      if b {
+        return true; 
+      } {
+        move;
+        return false;
+      }
+    );
+    grab
+  }
+robots:
```

Comment:
My solution, similar to the previous:
```
def until = \p. \c. b <- p; if b {} {c; until p c} end
def fwd = until blocked move end
build {turn west; until (ishere "goal") (fwd; turn right); grab}
```

Reply:
Weird, in REPL it works, but when I `--run` it, the robot stops in the wrong direction at the start.

Reply:
Did you have `turn west` or did you copy-paste the `turn right` from the previous solution?  ...it shouldn't matter though...

Reply:
Weird. I added a logger and a command to `view` the built robot, and then executed it using `run`, and got an error about needing to install a `branch predictor`.  Then in the same REPL session, I simply hit "up" and executed the exact same `run` command again... and the second time, it worked. :fearful:

Reply:
To be more specific, the robot is getting built, and it executes the `turn west`, and then when it goes to execute the `until` it fails on the runtime capability check for `if`, because it indeed does not have a `branch predictor` installed.  (It has a `compass`, `grabber`, `logger`, `scanner`, `solar panel`, and `treads`, but no branch predictor).

Reply:
Oh, so that is what it was. 😮 

Thanks for debugging it @byorgey. 👍

Reply:
I used your solution @byorgey with `require` as a temporary workaround. :+1: